tools/autograd/gen_variable_type.py

```diff
@@ -350,6 +350,18 @@
 return c10::Dispatcher::singleton().redispatch<${ret_and_arg_types}>(${profiled_dispatch_args});
 """)
 
+
+# TraceType templates
+TRACE_DISPATCH_FIND_OP = CodeTemplate("""\
+static auto op = c10::Dispatcher::singleton().findSchema({"aten::${operator_name}", "${overload_name}"});
```

Comment:
Can we factor things to eliminate the duplication between these `TRACE_DISPATCH_FIND_OP` and `TRACE_DISPATCH_CALL_OP` and `PROFILE_DISPATCH_UNBOXED` above? I know that takes you outside the new functionality but it seems like it's worth doing.

Oh also! I think you can replace the `findSchema()/TORCH_INTERNAL_ASSERT()` with `findSchemaOrThrow()` - see e.g. [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/function_wrapper.py#L136)

Reply:
Switched to `findSchemaOrThrow()` and merged `TRACE_DISPATCH_FIND_OP` & `TRACE_DISPATCH_CALL_OP`. I didn't touch `PROFILE_DISPATCH_UNBOXED` because I was told @ilia-cher  is going to remove it (calling profiler from dispatcher instead).

`TRACE_DISPATCH_FIND_OP` / `TRACE_DISPATCH_CALL_OP` were initially separated because `auto result = ` might be inserted before `TRACE_DISPATCH_CALL_OP` when it doesn't return void / mutating self. Now I pass "auto result = " into the CodeTemplate - however, I figured it unconditionally strips the trailing whitespace in the value [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/code_template.py#L44), so it will look like "auto result =c10::Dispatcher::singleton()..." - correct but ugly :(